vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -106,6 +149,14 @@ protected Object readObjectId(BsonReader reader, DecoderContext ctx) {
 
   @Override
   protected Object readDateTime(BsonReader reader, DecoderContext ctx) {
```

Comment:
Well I suppose one justification for implementing it this way would be if the user took some JSON retrieved from Mongo via the Vert.x client and then stored it into another Mongo using the Mongo Java driver directly. In this case it would barf if the JSON contained an ISO date < epoch.